app/services/category_hashtag_data_source.rb

```diff
@@ -52,11 +52,43 @@ def self.search(guardian, term, limit)
   end
 
   def self.search_sort(search_results, term)
-    search_results
-      .select { |item| item.slug == term }
-      .sort_by { |item| item.text.downcase }
-      .concat(
-        search_results.select { |item| item.slug != term }.sort_by { |item| item.text.downcase },
-      )
+    if term.present?
+      search_results
+        .select { |item| item.slug == term }
+        .sort_by { |item| item.text.downcase }
+        .concat(
+          search_results.select { |item| item.slug != term }.sort_by { |item| item.text.downcase },
+        )
+    else
+      search_results.sort_by { |item| item.text.downcase }
+    end
+  end
+
+  def self.search_without_term(guardian, limit)
+    guardian_categories = Site.new(guardian).categories
+
+    category_id_sql = <<~SQL
+      SELECT category_id, MAX(posts.created_at)
+      FROM topics
+      INNER JOIN posts ON posts.topic_id = topics.id
+      WHERE topics.deleted_at IS NULL
+        AND posts.deleted_at IS NULL
+        AND posts.created_at > (NOW() - INTERVAL '2 WEEKS')
+        AND topics.category_id IN (:category_ids)
+      GROUP BY category_id
+      ORDER BY MAX(posts.created_at) DESC
+      LIMIT :limit
```

Comment:
I'm a litter worried about this query because the performance of the query degrades as the activity of a site increases as we have to join against many posts just to figure out which is the "last active" category. I'm not sure what the requirements are here but are we able to rely on columns like `Category#topic_count`, `Category#post_count` or `Category#lastest_post_id` or `Category#latest_topic_id` to achieve the same result?

Reply:
`latest_topic_id` or `latest_post_id` may work here, I will try that out, then we can join the categories on post/topic `created_at` and sort by that, thanks